src/backend/utils/misc/guc.c

```diff
@@ -5276,7 +5277,7 @@ find_option(const char *name, bool create_placeholders, int elevel)
 	 * By equating const char ** with struct config_generic *, we are assuming
 	 * the name field is first in config_generic.
 	 */
-	res = (struct config_generic **) bsearch((void *) &key,
+	res = (struct config_generic **) bsearch((void *) &ptr,
```

Comment:
How can I reprocude the warning fixed here? GCC 10 and 13 don't find it

Reply:
```
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wimplicit-fallthrough=3 -Wvla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-but-set-variable -Wno-format-truncation -Wno-stringop-truncation -O3  -Werror -I. -I. -I../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -DPG_KRB_SRVTAB='"FILE:/usr/local/gpdb/etc/postgresql/krb5.keytab"'  -c -o guc.o guc.c
In function ‘guc_var_compare’,
    inlined from ‘bsearch’ at /usr/include/x86_64-linux-gnu/bits/stdlib-bsearch.h:33:23,
    inlined from ‘find_option’ at guc.c:5279:35:
guc.c:5321:38: error: array subscript ‘const struct config_generic[0]’ is partly outside array bounds of ‘const char[8]’ [-Werror=array-bounds=]
 5321 |         return guc_name_compare(confa->name, confb->name);
      |                                 ~~~~~^~~~~~
guc.c: In function ‘find_option’:
guc.c:5267:25: note: object ‘name’ of size 8
 5267 | find_option(const char *name, bool create_placeholders, int elevel)
      |             ~~~~~~~~~~~~^~~~
cc1: all warnings being treated as errors
make[4]: *** [<builtin>: guc.o] Error 1
make[4]: Leaving directory '/home/evgeniy/gpdb/adb-7.x/src/backend/utils/misc'
make[3]: *** [../../../src/backend/common.mk:39: misc-recursive] Error 2
make[3]: Leaving directory '/home/evgeniy/gpdb/adb-7.x/src/backend/utils'
make[2]: *** [common.mk:39: utils-recursive] Error 2
make[2]: Leaving directory '/home/evgeniy/gpdb/adb-7.x/src/backend'
make[1]: *** [Makefile:45: all-backend-recurse] Error 2
make[1]: Leaving directory '/home/evgeniy/gpdb/adb-7.x/src'
make: *** [GNUmakefile:11: all-src-recurse] Error 2
evgeniy@evgeniy-pc:~/gpdb/adb-7.x$ gcc --version
gcc (Ubuntu 13.2.0-4ubuntu3) 13.2.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```